core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -347,6 +347,23 @@ default boolean isComplete() {
      */
     CompletableFuture<Void> whenComplete();
 
+    /**
+     * Drain all data using {@link NoopSubscriber}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<Integer> source = StreamMessage.of(1, 2, 3);
+     * List<Integer> collected = new ArrayList<>();
+     * CompletableFuture<Void> future = source.peek(collected::add).subscribe();
+     * future.join();
+     * assert collected.equals(List.of(1, 2, 3));
+     * assert future.isDone();
+     * }</pre>
+     */
+    default CompletableFuture<Void> subscribe() {
+        subscribe(NoopSubscriber.get());
+        return NoopSubscriber.get().whenSubscribed();
```

Comment:
Wouldn't it be simpler to return `whenComplete()` here instead?
```suggestion
        return whenComplete();
```